### PR TITLE
Add foreign key system check

### PR DESCRIPTION
**Pull request by joshuaruesweg**
See #2977

### CODE REVIEW

File: wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php [hunk 1]
```diff
@@ -184,7 +219,23 @@ protected function validateMysql() {
 			}
 		}
 		
-		if ($this->results['mysql']['result'] && $this->results['mysql']['innodb']) {
+		// validate foreign keys
+		$this->results['mysql']['foreignKeys'] = true;
+		foreach ($this->foreignKeys as $table => $keys) {
+			$sql = "SHOW CREATE TABLE ". $table;
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php]: Use `information_schema`.

File: wcfsetup/install/files/acp/templates/systemCheck.tpl [hunk 2]
```diff
@@ -129,6 +129,18 @@
...
 		</dd>
 	</dl>
+	
+	<dl{if !$results[mysql][foreignKeys]} class="formError"{/if}>
+		<dt>{lang}wcf.acp.systemCheck.mysql.foreignKeys{/lang}</dt>
+		<dd>
+			{if $results[mysql][foreignKeys]}
+				{@$statusOk} {lang}wcf.acp.systemCheck.pass{/lang}
+			{else}
+				{@$statusInsufficient} {lang}wcf.acp.systemCheck.notSupported{/lang}
```
Comment (TimWolla) [inline wcfsetup/install/files/acp/templates/systemCheck.tpl]: Consider giving a more helpful message here: It's not that foreign keys are _unsupported_. The issue is that they (somehow) went missing.

File: wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php (hunk 1 above)
Comment (joshuaruesweg) [reply wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php]: Just a note: Using the `INFORMATION_SCHEMA` is much, much slower then the `SHOW CREATE TABLE` solution.

File: wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php [hunk 3]
```diff
@@ -184,7 +219,33 @@ protected function validateMysql() {
...
+				$innerConditionBuilder = new PreparedStatementConditionBuilder(false);
+				$innerConditionBuilder->add('REFERENCED_TABLE_SCHEMA = ?', [WCF::getDB()->getDatabaseName()]);
+				$innerConditionBuilder->add('REFERENCED_TABLE_NAME = ?', [$reference['referenceTable']]);
+				$innerConditionBuilder->add('REFERENCED_COLUMN_NAME = ?', [$reference['referenceColumn']]);
+				$innerConditionBuilder->add('TABLE_NAME = ?', [$table]);
+				$innerConditionBuilder->add('COLUMN_NAME = ?', [$column]);
+				
+				$conditionBuilder->add('('. $innerConditionBuilder .')', $innerConditionBuilder->getParameters());
+				
+				$keyCount++;
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php]: Possibly call this `expectedForeignKeyCount`.

**Pull request merged**